packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx

```diff
@@ -0,0 +1,244 @@
+
+    // should not be opened by default
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    // should open after 400ms
+    await act(async () => {
+      jest.advanceTimersByTime(399);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close
+    fireEvent.mouseLeave(getByRole('button', { name: buttonText }));
+    await act(async () => {
+      jest.advanceTimersByTime(999 + animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should open/close on focus/blur', async () => {
+    const buttonText = 'Hover me';
+    const { getByRole } = renderWithTheme(
+      <Tooltip content="Hello world">
+        <Button>{buttonText}</Button>
+      </Tooltip>,
+    );
+    await waitForPosition();
+
+    fireEvent.focus(getByRole('button', { name: buttonText }));
+
+    // on focus it should immediately open
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close
+    fireEvent.blur(getByRole('button', { name: buttonText }));
+    await act(async () => {
+      jest.advanceTimersByTime(animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should open/close on hovering over with non-interactive trigger element', async () => {
+    const tooltipContent = 'Hello world';
+    const { getByTestId } = renderWithTheme(
+      <Tooltip content={tooltipContent} shouldWrapChildren>
+        <InfoIcon color="surface.action.icon.default.highContrast" size="medium" />
+      </Tooltip>,
+    );
+    await waitForPosition();
+
+    const wrapper = getByTestId('tooltip-interactive-wrapper');
+    fireEvent.mouseEnter(wrapper);
+
+    // should not be opened by default
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    // should open after 400ms
+    await act(async () => {
+      jest.advanceTimersByTime(399);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close
+    fireEvent.mouseLeave(wrapper);
+    await act(async () => {
+      jest.advanceTimersByTime(999 + animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should open/close on focus/blur with non-interactive trigger element', async () => {
+    const { getByTestId } = renderWithTheme(
+      <Tooltip content="Hello world" shouldWrapChildren>
```

Comment:
Assuming we're going to update this with that TriggerWrapper right?

Reply:
Removed.